lib/dockerregistry/inventory.go

```diff
@@ -1973,6 +1976,60 @@ func (sc *SyncContext) RunChecks(
 	return nil
 }
 
+// Run is a function of ImageSizeCheck and checks that all
+// images to be promoted are under the max file size.
+func (check *ImageSizeCheck) Run() error {
+	const mbToByteShift = 20
+	maxImageSizeByte := check.MaxImageSize << mbToByteShift
+	oversizedImages := make([]string, 0)
+	invalidImages := make([]string, 0)
+	for edge := range check.PullEdges {
+		imageSize := check.DigestImageSize[edge.Digest]
+		if imageSize > maxImageSizeByte {
+			oversizedImages = append(oversizedImages,
+				string(edge.DstImageTag.ImageName))
+		}
+		if imageSize <= 0 {
+			invalidImages = append(invalidImages,
+				string(edge.DstImageTag.ImageName))
+		}
+	}
+
+	errStr := ""
+	if len(oversizedImages) > 0 {
+		sort.Strings(oversizedImages)
+		errStr += fmt.Sprintf("The following images were over the max file "+
```

Comment:
Better to display the actual image size for the offending image (probably in bytes but also in MiB).